dependencyPacker/build.gradle.kts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+import org.jetbrains.kotlin.konan.properties.KonanPropertiesLoader
+import org.jetbrains.kotlin.konan.util.DependencyProcessor
+import org.jetbrains.kotlin.konan.target.KonanTarget
+import org.jetbrains.kotlin.konan.target.Distribution
+import org.jetbrains.kotlin.konan.target.presetName
+
+/**
+ * Creates an archive with LLVM distribution without files
+ * that are not required for Kotlin/Native.
+ *
+ * For example, it excludes binaries that are part of LLVM testing infrastructure.
+ */
+inline fun <reified T: AbstractArchiveTask> createLlvmPackingTask(
+        host: KonanTarget,
+        whitelist: File?,
+        blacklist: File?,
+        crossinline additionalConfiguration: T.() -> Unit
+) {
+    val distribution = Distribution(
+            konanHomeOverride = rootProject.projectDir.absolutePath
+    )
+    val reducedLlvmAppendix = distribution.properties
+            .getProperty("reducedLlvmAppendix")
+    val propertiesLoader = object : KonanPropertiesLoader(
+            host = host,
+            target = host,
+            properties = distribution.properties,
+            baseDir = DependencyProcessor.defaultDependenciesRoot.absolutePath
+    ) {}
+
+    val hostName = host.presetName.capitalize()
+    val downloadTaskName = "downloadLlvmFor$hostName"
+
+    tasks.create(downloadTaskName) {
+        doFirst {
+            propertiesLoader.downloadDependencies()
+        }
+    }
+    tasks.create<T>("packLlvmFor$hostName") {
+        dependsOn(downloadTaskName)
+        additionalConfiguration()
+        description = "Packs LLVM dependency for $hostName into archive"
```

Comment:
(minor) You can specify a group to a task to make it visible in `./gradlew tasks`, e.g. `group = "Distribution packing"`